### ISSUE TITLE
Comment Loop: Comments displayed on password protected posts

### ISSUE THREAD

**Issue by peterwilsoncc** [WordPress/gutenberg#40747]
### Description

When a password protected post is created, the comment query loop will display comments prior to the password been entered. 

In the legacy comment template, WordPress requires the password be entered before displaying comments. 

```php
if ( post_password_required() ) { ?>
		<p class="nocomments"><?php _e( 'This post is password protected. Enter the password to view comments.' ); ?></p>
	<?php
	return;
}
```
([source in wordpress-develop](https://github.com/WordPress/wordpress-develop/blob/6742d0d7a65e37f24c67eeaae373f4c086c277a5/src/wp-includes/theme-compat/comments.php#L23-L27))

### Step-by-step reproduction instructions

1. Configure: WP 6.0 with TT1 theme
2. Create and publish a post (don't password protect it yet)
3. Visit the post and leave a comment
4. Edit the post and password protect it
5. Visit the post in a private/incognito browser window but _do not enter_ the password
6. observe the comments are not shown
7. Activate TT2 and configure comments to use the new comment loop
8. Visit the post in a private/incognito browser window but _do not enter_ the password
9. observe the comments are shown

### Screenshots, screen recording, code snippet

**Twenty Twenty-One**
![Screen Shot 2022-04-30 at 11 17 33 am](https://user-images.githubusercontent.com/519727/166085008-761c2640-6732-4502-a192-feb7ed3d4a8a.png)

**Twenty Twenty-Two**
![Screen Shot 2022-04-30 at 11 11 41 am](https://user-images.githubusercontent.com/519727/166085019-ffd90d8b-b619-4ce9-b2c5-5a7975ac2f05.png)

### Environment info

- WordPress 6.0/trunk - https://github.com/WordPress/wordpress-develop/commit/84be1644cd95d99d15f454f5803ad365eba499e6
- Gutenberg inactive
- Browser N/A

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by annezazu** [comment]
cc @SantosGuillamot and @ockham for good measure.

**Comment by peterwilsoncc** [comment]
This is the only solution I could find that doesn't require a major refactor of the comment query loop. 

While it does avoid having to modify the output of all the server side rendered inner blocks, it doesn't seem optimal from the perspective of how these things have been managed in the past.

In the experiment below, I decided not to display any content in the style of the screenshot for Twenty Twenty-One rather than display a message about entering a password. The post's content is already replaced with a password form so doubling up isn't required.

```php
add_filter(
	'render_block',
	function ( $block_content, $block ) {
		// Closure because I am hacking away...
		if ( 'core/comments-query-loop' !== $block['blockName'] ) {
			return $block_content;
		}

		if ( post_password_required() ) {
			return '';
		}

		return $block_content;
	},
	10,
	2
);
```

**Comment by ockham** [comment]
Fixed by #40755. Thanks a lot @peterwilsoncc!
